zerver/tests/test_openapi.py

```diff
@@ -1101,3 +1101,37 @@ def test_validator(self) -> None:
         validate_request(
             "/dev_fetch_api_key", "post", {}, {}, False, "200", intentionally_undocumented=True
         )
+
+
+class APIDocsSidebarTest(ZulipTestCase):
+    def test_link_in_sidebar(self) -> None:
+        """
+        Test to make sure that links of API documentation pages exist
+        in the sidebar and have the same label as the summary of the endpoint.
+        """
+        # These endpoints are in zulip.yaml, but not the actual docs.
+        exempted_docs = {
+            # (No /api/v1/ or /json prefix).
+            "get-file-temporary-url",
+            # This one is not used by any clients and is likely to get
+            # deprecated.
+            "update-subscriptions",
+            # This is rendered on the "Outgoing webhooks" page and hence is not
+            # linked in the sidebar.
+            "zulip-outgoing-webhooks",
+        }
+        sidebar_path = "api_docs/sidebar_index.md"
+        rest_endpoints_path = "api_docs/include/rest-endpoints.md"
+        with open(sidebar_path) as fp:
+            sidebar_content = fp.read()
+        with open(rest_endpoints_path) as fp:
+            sidebar_content += fp.read()
+
+        paths = openapi_spec.openapi()["paths"]
+        for endpoint in paths:
+            for method in paths[endpoint]:
+                operationId = paths[endpoint][method].get("operationId")
+                summary = paths[endpoint][method].get("summary")
+                if operationId and operationId not in exempted_docs:
+                    link = f"[{summary}](/api/{operationId})"
+                    assert link in sidebar_content
```

Comment:
How much time does this test take to run? I worry a bit about it doing quadratic work.

Reply:
It takes about 1 second to run. Yeah, it has quadratic time complexity but there are not many endpoints for it to scale. Do you want me to do this in linear time?

Reply:
Depending how you measured it, a second of actual test runtime could be fairly bad or just the per-test overhead. I think it'd be pretty reasonable to do it in linear time, so we might as well just do that so we don't have to think about.

(Can you just make a `Set` of the lines of the sidebar content?)

Reply:
> Can you just make a Set of the lines of the sidebar content?
OK, I have done that.